app/src/main/java/io/gnosis/safe/ui/settings/app/passcode/EnterPasscodeFragment.kt

```diff
@@ -54,9 +56,16 @@ class EnterPasscodeFragment : BaseViewBindingFragment<FragmentPasscodeBinding>()
 
     override fun onResume() {
         super.onResume()
-        binding.input.setRawInputType(InputType.TYPE_CLASS_NUMBER)
-        binding.input.delayShowKeyboardForView()
-        authenticateWithBiometrics()
+        if (alreadyStarted) {
+            findNavController().popBackStack(R.id.enterPasscodeFragment, true)
```

Comment:
could you elaborate what happens here exactly and why are we removing fragment from stack? Is it possible for EnterPasscodeFragment to be added multiple times?

Reply:
you could use nav options  to prevent such problem if this is the case here
```
 navigate(
    @NonNull NavDirections directions,
    @Nullable NavOptions navOptions
)
```

```
NavOptions.Builder().setLaunchSingleTop(true).build() 
```

Reply:
It happenned because each time the app is pulled into foreground, appInForeground() is called an it starts a new enterPasscodeFragment.

Reply:
And since we have no feedback when it is closed, I tried to prevent multiple openings here. But the flag launchSingleTop could also work.